Sources/HTTP/Program/Program.swift

```diff
@@ -1,22 +1,46 @@
 import URI
 import Transport
 
+private var _defaultMiddlewareStorage: [String: [Middleware]] = [:]
+
 public protocol Program {
     var host: String { get }
     var port: Int { get }
     var securityLayer: SecurityLayer { get }
-    init(host: String, port: Int, securityLayer: SecurityLayer) throws
+    var middleware: [Middleware] { get }
+    init(host: String, port: Int, securityLayer: SecurityLayer, middleware: [Middleware]) throws
+    static var defaultMiddleware: [Middleware] { get set }
+}
+
+extension Program {
+    public static var defaultMiddleware: [Middleware] {
+        get {
+            let name = "\(type(of: Self.self))"
+            guard let middleware = _defaultMiddlewareStorage[name] else {
+                _defaultMiddlewareStorage[name] = []
+                return []
+            }
+
+            return middleware
+        }
+
+        set {
+            let name = "\(type(of: Self.self))"
+            _defaultMiddlewareStorage[name] = newValue
+        }
+    }
 }
 
 extension Program {
     public static func make(
         host: String? = nil,
         port: Int? = nil,
-        securityLayer: SecurityLayer = .tls(nil)
+        securityLayer: SecurityLayer = .tls(nil),
+        middleware: [Middleware] = []
     ) throws -> Self {
         let host = host ?? "0.0.0.0"
         let port = port ?? securityLayer.port()
-        return try Self(host: host, port: port, securityLayer: securityLayer)
+        return try Self(host: host, port: port, securityLayer: securityLayer, middleware: Self.defaultMiddleware + middleware)
```

Comment:
@tannernelson No, because this would override the default, but if the initializer is called outside of this function, it won't be added